lms/static/js/bookmarks/collections/bookmarks.js

```diff
@@ -1,20 +1,24 @@
 ;(function (define) {
     'use strict';
-    define(['backbone', 'js/bookmarks/models/bookmark'],
-        function (Backbone, BookmarkModel) {
+    define(['backbone', 'common/js/components/collections/paging_collection', 'js/bookmarks/models/bookmark'],
+        function (Backbone, PagingCollection, BookmarkModel) {
 
+            return PagingCollection.extend({
+                initialize: function(options) {
+                    PagingCollection.prototype.initialize.call(this);
 
-        return Backbone.Collection.extend({
-            model: BookmarkModel,
+                    this.course_id = options.course_id;
+                    this.server_api.course_id = function () { return encodeURIComponent(this.course_id); };
+                    this.server_api.fields = function () { return encodeURIComponent('display_name,path'); };
+                    delete this.server_api.sort_order; // Sort order is not specified for the Bookmark API
+                },
 
-            url: function() {
-                return $(".courseware-bookmarks-button").data('bookmarksApiUrl');
-            },
+                model: BookmarkModel,
 
-            parse: function(response) {
-                return response.results;
-            }
-        });
+                url: function() {
+                    return $(".courseware-bookmarks-button").data('bookmarksApiUrl');
```

Comment:
I'd like to see the url passed through the factory js. I don't think the DOM is the right place for this data to live.